internal/testutil/validator_addon_label.go

```diff
@@ -0,0 +1,45 @@
+package testutil
+
+import (
+	"github.com/mt-sre/addon-metadata-operator/api/v1alpha1"
+	"github.com/mt-sre/addon-metadata-operator/pkg/utils"
+	"github.com/mt-sre/addon-metadata-operator/pkg/validators"
+)
+
+type ValidatorAddonLabelTestBundle struct{}
+
+func (val ValidatorAddonLabelTestBundle) Name() string {
+	return "Addon Label Validator"
+}
+
+func (val ValidatorAddonLabelTestBundle) Run(mb utils.MetaBundle) (bool, error) {
+	return validators.ValidateAddonLabel(&mb)
```

Comment:
Unrelated to this PR, but I noticed that we're passing pointers to the validation fns. I think thats unnecessary and could cause problems with other validations, if someone accidentally modifies the metabundle object. Each validation fn should receive its own private copy of the metabundle object.

Reply:
Bingo! Even I thought so while coding out this PR and have already a dedicated branch/commit ready locally shifting from pointer semantics to value semantics in the validator functions :)

PS: Nice catch though, even I prefer complying with value semantics instead of pointer semantics wherever possible in a consistent manner as pointer receivers (stored on heap) end up imposing more pressure on the GC as compared to value receivers (stored on stack) 🚀

Reply:
https://github.com/mt-sre/addon-metadata-operator/pull/17